app/admin/missions.rb

```diff
@@ -74,6 +76,33 @@
     end
   end
 
+  controller do
+    def update
+      if update_transaction.success?
+        flash[:notice] = translate 'activerecord.notices.messages.update_success'
+        redirect_to admin_mission_path(resource.id)
+      else
+        flash[:error] = update_transaction.failure
+        render :edit
+      end
+    end
+
+    private
+
+    def update_transaction
+      @update_transaction ||=
+        begin
```

Comment:
Why do you need a `begin` here?

Reply:
see https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/
in multi-line memoization section